src/leiningen/monolith.clj

```diff
@@ -274,28 +275,68 @@
     (fingerprint/clear project opts more)))
 
 
+(defn ^:higher-order with-dependency-set
+  "Run a task with a set of managed dependencies from a named dependency set.
+
+   Overrides the dependencies from the named dependency set into the project.
+   For the root project, this means the managed dependencies will be overwritten
+   with the dependencies from the named set. For subprojects, the
+   `:monolith/dependency-set` metadata key will be set to the named set. If the
+   project does not have the `:monolith/dependency-set` key defined, it will be
+   added.
+
+   For combining with other higher-order tasks, this task should be the last
+   evaluated.
+
+   A gotcha: this task does not update the project's `project.clj` file, so the
+   dependencies will not be saved to disk. This is intentional, as the task is
+   meant to be used for temporary operations that need to be run with a specific
+   set of dependencies. This also means that the task will not work with tasks
+   that require the project file to be updated.
+
+   Options:
+     :only    Use _only_ the dependencies from the named set. This replaces
```

Comment:
I'm wondering if this is actually useful now, given that it could basically never do something helpful in a subproject. 🤔 It seems like if you _only_ want the managed dependencies, you should run the `with-dependency-set` task in the metaproject? If a user did have a case where they wanted to remove actual dependencies from consideration, they could use `update-in`:
```
% lein update-in :dependencies empty -- pprint :dependencies
()
```

Reply:
We have a use case for checking outdated dependencies for just a dependency set where this is useful, but I could go either way on it honestly. It's the difference between
```
lein monolith with-dependency-set :only :foo antq
```
and
```
lein update-in :dependencies empty -- monolith monolith with-dependency-set :only :foo antq
```

I also didn't test this behavior but I'll take that up.

Reply:
Right, I know about the just-the-dependency-set case, but I'm asking if that's not already solved by running the `with-dependency-set` command in the metaproject itself, where there are no dependencies defined (except Clojure itself).

Reply:
Ah, sorry should have clarified. The middleware will pick up any dependencies in your user profile as well, and we want to ignore those.

Reply:
We do also, for example, have `org.clojure/clojure` as a top level dependency in the metaproject

Reply:
Ah, I see - is the concern there in automated use-cases, or a human running this locally? Automated tools shouldn't have a `:user` profile to worry about, so maybe that's enough of an answer.

For humans, this is far from the only tool where user-level config leaking in can change the results. I have some concern with trying to play whack-a-mole here, because even if we override `:dependencies` with this option, there are lots of plugins a user might have which add their own dependencies to the project as part of _their_ middleware, and then ordering of the plugins running matters to the end result. It seems safer to give the standard advice to fully remove the user profile for reproducible builds using `lein with-profile -user ...`.

Reply:
That's fair. I'll remove it and update the docs